cscs-checks/microbenchmarks/cpu/roofline/berkeley-ert.py

```diff
@@ -0,0 +1,444 @@
+        'mpi_tasks': [6, 12],
+        # NOTE: uncomment for full search:
+        # 'mpi_tasks': [1, 2, 3, 4, 6, 8, 12],
+        # NOTE: lscpu: L1d cache: 32K L1i cache: 32K
+        #               L2 cache: 256K L3 cache: 30720K
+    },
+    'BWL': {
+        'ref_GFLOPs': 330.0,
+        'ref_DRAMbw': 53.0,
+        'mpi_tasks': [18, 36],
+        # NOTE: uncomment for full search:
+        # 'mpi_tasks': [1, 2, 3, 4, 6, 9, 12, 18, 36],
+        # NOTE: lscpu: L1d cache: 32K L1i cache: 32K
+        #               L2 cache: 256K L3 cache: 46080K
+    },
+}
+
+
+# {{{ class RunErt_Base
+class RunErt_Base(rfm.RegressionTest):
+    # {{{
+    """
+    This check runs and plots the ERT for different cpus.
+    The kernels are set as:
+    > grep 'define KERNEL' Empirical_Roofline_Tool-1.1.0/Kernels/kernel1.h
+    #define KERNEL1(a, b, c) ((a) = (b) + (c))
+    #define KERNEL2(a, b, c) ((a) = (a) * (b) + (c))
+
+    The flops are set in:
+    > egrep "add .* flop" Kernels/kernel1.h
+    #if (ERT_FLOP & 1) == 1 /* add 1 flop */
+    #if (ERT_FLOP & 2) == 2 /* add 2 flops */
+    #if (ERT_FLOP & 4) == 4 /* add 4 flops */
+    #if (ERT_FLOP & 8) == 8 /* add 8 flops */
+    #if (ERT_FLOP & 16) == 16 /* add 16 flops */
+    #if (ERT_FLOP & 32) == 32 /* add 32 flops */
+    #if (ERT_FLOP & 64) == 64 /* add 64 flops */
+    #if (ERT_FLOP & 128) == 128 /* add 128 flops */
+    #if (ERT_FLOP & 256) == 256 /* add 256 flops */
+    #if (ERT_FLOP & 512) == 512 /* add 512 flops */
+    #if (ERT_FLOP & 1024) == 1024 /* add 1024 flops */
+
+    A typical output is:
+    working_set_size*bytes_per_elem
+                           t
+                                     seconds  total_bytes  total_flops
+           1024            1         731.238         2048       131072
+           1024            2           7.705         4096       262144
+           1024            4          11.562         8192       524288
+           1024            8          22.242        16384      1048576
+           etc...
+    """
+    # }}}
+    def __init__(self):
+        self.descr = f'Empirical Roofline Toolkit (Base for building/running)'
+        # pe step
+        # build step
+        # run step
+        # postprocess step
+        # sanity step
+
+    # {{{ hooks
+    @rfm.run_before('compile')
+    def set_compiler_flags_and_variables(self):
+        self.sourcesdir = os.path.join(
+            self.current_system.resourcesdir,
+            'roofline',
+            'cs-roofline-toolkit.git',
+            'Empirical_Roofline_Tool-1.1.0',
+        )
+        self.readonly_files = [
+            'Batch',
+            'Config',
+            'Drivers',
+            'ert',
+            'ert_cscs.py',
+            'ERT_Users_Manual.pdf',
+            'Kernels',
+            'Plot',
+            'Python',
+            'README.md',
+            'Results',
+            'Scripts',
+        ]
+        self.sourcepath = 'Kernels/kernel1.cxx Drivers/driver1.cxx'
```

Comment:
Oook, looks like a hack this one :-) Why not a Makefile?

Reply:
I am trying to not add any change to the official repo

Reply:
Perhaps add a comment here to explain this?